src/OpenTK/DisplayDevice.cs

```diff
@@ -175,12 +146,7 @@ public DisplayResolution SelectResolution(int width, int height, int bitsPerPixe
         /// </summary>
         public IList<DisplayResolution> AvailableResolutions
         {
-            get { return available_resolutions_readonly; }
-            internal set
-            {
-                available_resolutions = (List<DisplayResolution>)value;
-                available_resolutions_readonly = available_resolutions.AsReadOnly();
-            }
+            get { return implementation.GetAvailableResolutions(id); }
         }
 
         /// <summary>Changes the resolution of the DisplayDevice.</summary>
```

Comment:
They should probably be the same, yes. In fact, it might be best if the property forwarded its setter to this method, so there's no duplication in how the resolution is set.